sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -117,9 +122,11 @@ class SparkSession private(
   @InterfaceStability.Unstable
   @transient
   lazy val sessionState: SessionState = {
-    SparkSession.reflect[SessionState, SparkSession](
-      SparkSession.sessionStateClassName(sparkContext.conf),
-      self)
+    parentSessionState
+      .map(_.copy(this))
```

Comment:
As this is a lazy val, the cloned `SparkSession` will not copy parent `SessionState` immediately. So if the parent `SessionState` is changed, e.g., new functions registered, before this lazy variable initialization, the cloned one will also have new functions.

 It will not match the description `...Changes to base session are not propagated to cloned session, cloned is independent after creation...` below.